### DISCUSSION TITLE
Retrieve custom collection with his associated filters

### DISCUSSION THREAD

**Discussion by Minirock** [Smile-SA/elasticsuite#2909] [Q&A]
Using Smile Elasticsuite on magento 2.4 : 

**Feature description**
We have a form where the user enter a data : 'plate_identifier"

This data is not part of the product data but is used to retreieve a list of  matching 'reference' product attribute. The thing is that there can be a huge amount of attribute matching.

I'm expecting to print that custom collection  (that i'm able to calculate by myself) on a custom listing page. 

**The issue i'm facing**

I've been trying 2 approaches : 

- The first one was to create a virtual category in backend, and to manually add all the 'reference' to the search query...as a result, it worked perfectly fine, i was getting all the products matching these references. But the issue was that it was generating too long url like 'http://website/custom-search-result?reference=1&reference=2&....&reference=xxx' which was resulting into error 500 as the url was more than 2048 character.
So i gave up that solution as it would have been bad in term of SEO anyway

- The second approach is to build the collection directly by myself and to create the layer / template too. As a result I do have a collection filter properly in frontend; but, sadly an other issue appears, the filter are not adapting themselves to the collection printed. I can't tell if that's due to a bug or a missconfiguration from me. The filter act like if all the products were used in the collection.

Here is a sample of the result

![image](https://user-images.githubusercontent.com/16558403/231481610-97734162-5faf-47a8-ab00-9e4e06bc38a8.png)

Only 4 results in the collection; but all filter items are being shown where for example i expect that the filter items attributes and their count would adapt to these 4 results. 

Any idea how to do this ?

**Comment by romainruaud** [comment]
Hi @Minirock 

that's something we already made on several Smile projects previously (Carrefour MonAuto, Avatacar, etc....).

Basically, you need to : 

- index "vehicles compatibility" on product level. I know there are several different fields to handle this (chassis, ktype, etc...).
- when people submit a plate_number, you fetch the chassis, ktype, etc... from the plate number
- then you add a filter that will be applied everywhere, like the stock filter : https://github.com/Smile-SA/elasticsuite/blob/2.11.x/src/module-elasticsuite-catalog/etc/elasticsuite_search_request.xml#L23
- this filter will inject proper query clauses according to chassis, ktype, etc... when browsing

I cannot say more as part of our Open Source support policy there, because that's very project specific.

I'm moving your issue as a Discussion.

Regards
